src/Text/Parsing/StringParser/String.purs

```diff
@@ -111,3 +116,38 @@ anyLetter = lowerCaseChar <|> upperCaseChar <?> "Expected a letter"
 -- | Match a letter or a number.
 alphaNum :: Parser Char
 alphaNum = anyLetter <|> anyDigit <?> "Expected a letter or a number"
+
+-- | Build the regular expression from the pattern and match it, ensuring
+-- | that the pattern only attempts to match from the start of the target.
+regex' :: String -> Parser String
+regex' pat =
+    case er of
+      Left _ ->
+        fail $ "Illegal regex " <> pat
+      Right r ->
+        regex r
+    where
+      pattern =
+        if startsWith "^" pat then
+          pat
+        else
+          "^" <> pat
+      er = Regex.regex pattern noFlags
+
+-- | Match the regular expression.
+regex :: Regex.Regex -> Parser String
+regex r =
+  Parser \{ str, pos } ->
+    let
+      remainder = drop pos str
+    in
+      -- reduce the possible array of matches to 0 or 1 elements to aid Array pattern matching
+      case uncons $ fromMaybe [] $ Regex.match r remainder of
+        Just { head: Just matched, tail: _ }  ->
+          -- only accept matches at position 0
+          if startsWith matched remainder then
+            Right { result: matched, suffix: { str, pos: pos + length matched } }
+          else
+            Left { pos, error: ParseError $ "no match - consider prefacing the pattern with '^'" }
```

Comment:
Maybe prefix the error with `Text.Parsing.StringParser.String.regex:` so it's clear where this error comes from? Same with the one in `regex'` above.